sdk/include/opentelemetry/sdk/logs/logger.h

```diff
@@ -67,9 +67,10 @@ class Logger final : public opentelemetry::logs::Logger
   // The name of this logger
   std::string logger_name_;
 
-  // The logger context of this Logger. Uses a weak_ptr to avoid cyclic dependency issues the with
-  std::weak_ptr<LoggerContext> context_;
+  // order of declaration is important here - instrumentation library should destroy after
+  // logger-context.
   std::unique_ptr<instrumentationlibrary::InstrumentationLibrary> instrumentation_library_;
+  std::shared_ptr<LoggerContext> context_;
```

Comment:
Is cyclic dependency still an issue here as it changes `weak_ptr` to `shared_ptr`?

Reply:
There is no cyclic dependency right now.It's only be hold in `Logger` and `LoggerProvider` and `Logger` is only hold by `LoggerProvider`.